### PR TITLE
chore(keyring-controller): deprecate QR Keyring methods

### PR DESCRIPTION
**Pull request by mikesposito**
## Explanation

With the introduction of `withKeyring`, `KeyringController` provides a solid alternative to QR-specific methods. This is a good chance to start moving away from them.

This PR deprecates all QR-specific methods, to discourage further usage in clients and promotes `withKeyring`.

## References

Related to [#3786](https://github.com/MetaMask/core/issues/3786)

### `@metamask/keyring-controller`

- **DEPRECATED**: Deprecated QRKeyring-specific methods from `KeyringController`

## Checklist

- [ ] I've updated the test suite for new or updated code as appropriate
- [ ] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [ ] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

**Review by mcmire — APPROVED** [review]
Nice, great news!

File: packages/keyring-controller/src/KeyringController.ts [hunk 1]
```diff
@@ -1523,41 +1532,94 @@ export class KeyringController extends BaseController<
...
+   */
   async resetQRKeyringState(): Promise<void> {
     (await this.getOrAddQRKeyring()).resetStore();
   }
 
+  /**
+   * Get QR keyring state.
+   *
+   * @returns Promise resolving to the keyring state.
+   * @deprecated Use `withKeyring` or subscribe to KeyringController:qrKeyringStateChange
```
Comment (ccharly) [inline packages/keyring-controller/src/KeyringController.ts:1549]:
```suggestion
   * @deprecated Use `withKeyring` or subscribe to `"KeyringController:qrKeyringStateChange"`
```

File: packages/keyring-controller/src/KeyringController.ts [hunk 2]
```diff
@@ -1523,41 +1532,94 @@ export class KeyringController extends BaseController<
...
     (await this.getOrAddQRKeyring()).cancelSync();
   }
 
+  /**
+   * Connect to QR hardware wallet.
+   *
+   * @param page - The page to connect to.
+   * @returns Promise resolving to the connected accounts.
+   * @deprecated Use of this method is discouraged as it creates a dangling promise
+   * internal to the QRKeyring, which can lead to unpredictable deadlocks. Please use
```
Comment (ccharly) [inline packages/keyring-controller/src/KeyringController.ts:1620]:
```suggestion
   * internal to the `QRKeyring`, which can lead to unpredictable deadlocks. Please use
```
Or
```suggestion
   * internal to the QR keyring, which can lead to unpredictable deadlocks. Please use
```

**Review by ccharly — COMMENTED** [review]
Nitpick: Mostly cosmetic suggestions to match with other comments (feel free to ignore them!) 😃

**Review by mcmire — APPROVED** [review]

**Pull request merged**